test/lib/containerManager.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+var path = require('path');
+var request = require('request');
+var should = require('should');
+var level = require('level');
+
+var Loader = require('yaml-config-loader');
+var CM = require('../../lib/ContainerManager');
+var server = new CM();
+var loader = new Loader();
+var db = level('/tmp/db', {db: require('memdown')});
+
+const port = 9876;
+const reapedBuildId = 'a38c39ed-3bb9-44ba-9050-8a4337384d3b';
+const pendingBuildId = 'c65c469a-60a9-43bb-848c-e96242e6b610';
+const missingBuildId = 'fedcda29-cf3b-4ff3-83ce-308a632d477b';
+
+loader.add(path.resolve(path.join(__dirname, '../..', 'defaults.yaml')));
+
+
+describe('Server', function() {
+
+  before('start server', function(done) {
+    var nock = require('nock');
+    nock.enableNetConnect('localhost:' + port);
+    loader.load(function(err, config) {
+      if (err) {
+        return done(err);
+      }
+      config.levelupDB = db;
+      config.port = port;
+      config.api.token = 'testToken';
+
+      db.put(`builds!${reapedBuildId}`, JSON.stringify({container: true, reaped: true}));
+      db.put(`builds!${pendingBuildId}`, JSON.stringify({container: false, reaped: false}));
+      server.configure(config, function(err) {
+        if (err) {
+          return done(err);
+        }
+        server.run({config: config}, function(err) {
+          if (err) {
+            return done(err);
+          }
+          return done();
+        });
+      });
+    });
+  });
+
+  after('Stop server', function(done) {
+    server = null;
+    done();
+  });
+
+  it('should return an error if the build id is invalid', function(done) {
+    const badBuildId = '38c39ed-3bb9-44ba-9050-8a4337384d3b';
+    var opts = {
+      url: `http://localhost:${port}/container/proxy?build=${badBuildId}`,
+      method: 'POST',
+      headers: {
+        authorization: 'Bearer testToken',
+      },
```

Comment:
This test does not set the content type accept header like the others. Is there a reason why? Also, it could be a little simpler to setup request defaults, and reuse the defaults in each specific test.